autoscheduler/frontend/src/components/SchedulingPage/OptionsCard/OptionsCard.tsx

```diff
@@ -0,0 +1,88 @@
+import * as React from 'react';
+import {
+  Button, Checkbox, ListItem, ListItemIcon, ListItemText,
+} from '@material-ui/core';
+import { useDispatch, useSelector } from 'react-redux';
+import GenericCard from '../../GenericCard/GenericCard';
+import * as styles from './OptionsCard.css';
+import { replaceSchedules } from '../../../redux/actions/schedules';
+import selectSchedule from '../../../redux/actions/selectedSchedule';
+import Meeting from '../../../types/Meeting';
+// DEBUG
+import fetch from './generateSchedulesMock';
+import { RootState } from '../../../redux/reducer';
+import { CourseCardArray } from '../../../types/CourseCardOptions';
+import Availability from '../../../types/Availability';
+import { formatTime } from '../../../timeUtil';
+
+const OptionsCard: React.FC = () => {
+  const [includeFull, setIncludeFull] = React.useState(false);
+  const courseCards = useSelector<RootState, CourseCardArray>((state) => state.courseCards);
+  const avsList = useSelector<RootState, Availability[]>((state) => state.availability);
+  const dispatch = useDispatch();
+  const fetchSchedules = React.useCallback(() => {
+    // make courses object
+    const courses = [];
+    for (let i = 0; i < courseCards.numCardsCreated; i++) {
+      if (courseCards[i] && courseCards[i].course) {
+        courses.push({
+          subject: courseCards[i].course.split(' ')[0],
+          courseNum: courseCards[i].course.split(' ')[1],
```

Comment:
```suggestion
        const [subject, courseNum] = courseCards[i].course.split(' ');
        courses.push({
          subject,
          courseNum,
```
Avoids creating the array from `split` twice